config/index.js

```diff
@@ -25,20 +33,37 @@ function printConfig (config) {
 *    ]
 *
 */
-module.exports = function (config, overrides) {
-  config.load = function () {
+module.exports = function (config, configDirPath, overrides) {
+  if (!overrides && Array.isArray(configDirPath)) {
+    overrides = configDirPath
+    configDirPath = null
+  }
+
+  assert(!overrides || Array.isArray(overrides),
+    'if present, overrides must be an array')
+
+  config.load = async function () {
     if (config.applyDefaults) {
       config.applyDefaults()
     }
 
+    let allOverrides = []
+    if (configDirPath) {
+      const serviceEnv = BaseServiceConfig.getServiceEnvironment()
+      const env = (serviceEnv && serviceEnv.toLowerCase()) || 'localhost'
+      allOverrides = [ await confugu.load(`${configDirPath}/${env}.yml`) ]
+    }
+
     if (overrides) {
-      overrides.forEach((override) => {
-        for (const key in override) {
-          config.set(key, override[key])
-        }
-      })
+      allOverrides = [...allOverrides, ...overrides]
```

Comment:
Good recommendation - looks much better now. Though, I did still use an object to build from rather than unshifting and assigning to avoid some extra conditional messiness (and it seemed a little less readable to me).